src/integrationTest/java/org/opensearch/test/framework/audit/AuditLogsRule.java

```diff
@@ -122,16 +135,35 @@ private void whenTimeoutOccurs(String methodName) {
     private void afterTest() {
         TestRuleAuditLogSink.unregisterListener();
         this.currentTestAuditMessages = null;
+        this.currentTransportTestAuditMessages = null;
     }
 
     private void beforeTest(String methodName) {
         log.info("Start collecting audit logs before test {}", methodName);
         this.currentTestAuditMessages = synchronizedList(new ArrayList<>());
+        this.currentTransportTestAuditMessages = synchronizedList(new ArrayList<>());
         TestRuleAuditLogSink.registerListener(this);
     }
 
     public void onAuditMessage(AuditMessage auditMessage) {
-        currentTestAuditMessages.add(auditMessage);
-        log.debug("New audit message received '{}', total number of audit messages '{}'.", auditMessage, currentTestAuditMessages.size());
+        if (auditMessage.getAsMap().keySet().contains("audit_transport_headers")) {
+            if (log.isDebugEnabled()) {
+                log.debug(
+                    "New transport audit message received '{}', total number of transport audit messages '{}'.",
+                    auditMessage,
+                    currentTransportTestAuditMessages.size()
+                );
+            }
+            currentTransportTestAuditMessages.add(auditMessage);
+        } else {
```

Comment:
nit: the only diff I see in this if-else block is the word `transport` in the log message. Can we extract and modify to pass message depending on the condition?

Reply:
@DarshitChanpura there is also a different list used in each parts of if-else statement
I could refactor it into something like this:
```
public void onAuditMessage(AuditMessage auditMessage) {
    List<AuditMessage> targetList;
    String logMessage;

    if (auditMessage.getAsMap().containsKey("audit_transport_headers")) {
        targetList = currentTransportTestAuditMessages;
        logMessage = "New transport audit message received '{}', total number of transport audit messages '{}'.";
    } else {
        targetList = currentTestAuditMessages;
        logMessage = "New audit message received '{}', total number of audit messages '{}'.";
    }

    if (log.isDebugEnabled()) {
        log.debug(logMessage, auditMessage, targetList.size());
    }

    targetList.add(auditMessage);
}
``` 
It's little bit harder to read/follow what's happening so not sure about this one :/
Let me know what do you think